vcr/api/v2/holder.go

```diff
@@ -0,0 +1,169 @@
+}
+
+// searchType returns the first type that is in this list:
+// - NutsOrganizationCredential
+// - NutsAuthorizationCredential
+func (ch typeHelper) searchType() string {
+	for _, c := range ch.Type {
+		switch c {
+		case credential.NutsOrganizationCredentialType:
+			fallthrough
+		case credential.NutsAuthorizationCredentialType:
+			return c
+		}
+	}
+	return ""
+}
+
+// SearchVCs checks the context used in the JSON-LD query, based on the contents it maps to a non-JSON-LD query
+// After V1, this needs to be remapped to a DB search that supports native JSON-LD
+func (w *Wrapper) SearchVCs(ctx echo.Context, params SearchVCsParams) error {
+	var rawRequest typeHelper
+
+	// we parse without bind to bypass the defaults for Content-Type (application/ld+json)
+	bodyBytes, err := ioutil.ReadAll(ctx.Request().Body)
+	if err != nil {
+		return core.InvalidInputError("failed to parse request body: %w", err)
+	}
+
+	err = json.Unmarshal(bodyBytes, &rawRequest)
+	if err != nil {
+		return core.InvalidInputError("failed to unmarshall JSON request body: %w", err)
+	}
+
+	untrusted := false
+	if params.Untrusted != nil {
+		untrusted = *params.Untrusted
+	}
+
+	searchContext := rawRequest.searchType()
+	switch searchContext {
+	case credential.NutsAuthorizationCredentialType:
+		return w.searchAuths(ctx, untrusted, bodyBytes)
+	case credential.NutsOrganizationCredentialType:
+		return w.searchOrgs(ctx, untrusted, bodyBytes)
+	}
+
+	return core.InvalidInputError("given type not supported")
+}
+
+func (w *Wrapper) searchOrgs(ctx echo.Context, allowUntrusted bool, body []byte) error {
+	// some helper structs
+	type helper struct {
+		CredentialSubject struct {
+			ID           string `json:"id"`
+			Organization struct {
+				Name string `json:"name"`
+				City string `json:"city"`
+			} `json:"organization"`
+		} `json:"credentialSubject"`
+	}
+
+	ldQuery := helper{}
+	if err := json.Unmarshal(body, &ldQuery); err != nil {
+		return core.InvalidInputError("failed to unmarshall JSON request body: %w", err)
+	}
+
+	// convert to query
+	query, err := w.VCR.Registry().QueryFor(concept.OrganizationConcept)
+	if err != nil {
+		return err
+	}
+	if ldQuery.CredentialSubject.Organization.Name != "" {
+		query.AddClause(concept.Prefix("organization.name", ldQuery.CredentialSubject.Organization.Name))
+	}
+	if ldQuery.CredentialSubject.Organization.City != "" {
+		query.AddClause(concept.Prefix("organization.city", ldQuery.CredentialSubject.Organization.City))
+	}
+	if ldQuery.CredentialSubject.ID != "" {
+		query.AddClause(concept.Eq("subject", ldQuery.CredentialSubject.ID))
+	}
+
+	results, err := w.VCR.Search(ctx.Request().Context(), query, allowUntrusted, nil)
+	if err != nil {
+		return err
+	}
+
+	return ctx.JSON(http.StatusOK, results)
+}
+
+func (w *Wrapper) searchAuths(ctx echo.Context, allowUntrusted bool, body []byte) error {
+	// some helper structs that reflect current required query params
+	type helper struct {
+		CredentialSubject struct {
```

Comment:
this will all go when v1 API is removed, right?

Reply:
the underlying store (go-leia) must create indices based on JSON-LD instead of the current templates. Then the JSON-LD documents can be routed straight into go-leia without any conversion.
This PR is to provide a stable API for devs. Then in a next version we can change the underlying DB.

Reply:
see #821 for v2 milestone